### PR TITLE
fix: update packages to latest versions and add basic unit tests for the project

### PR DESCRIPTION
**Pull request by Joel-hanson**
- Adding new 19 tests

```bash
[INFO] --------< com.ibm.eventstreams.connect:kafka-connect-jdbc-sink >--------
[INFO] Building kafka-connect-jdbc-sink 0.0.2
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The artifact org.slf4j:slf4j-log4j12:jar:2.0.7 has been relocated to org.slf4j:slf4j-reload4j:jar:2.0.7
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ kafka-connect-jdbc-sink ---
[WARNING] The artifact xml-apis:xml-apis:jar:2.0.2 has been relocated to xml-apis:xml-apis:jar:1.0.b2
[INFO] com.ibm.eventstreams.connect:kafka-connect-jdbc-sink:jar:0.0.2
[INFO] +- org.postgresql:postgresql:jar:42.6.0:compile
[INFO] |  \- org.checkerframework:checker-qual:jar:3.31.0:runtime
[INFO] +- com.ibm.db2:jcc:jar:11.5.8.0:compile
[INFO] +- org.apache.kafka:connect-api:jar:3.4.0:compile
[INFO] |  \- javax.ws.rs:javax.ws.rs-api:jar:2.1.1:runtime
[INFO] +- org.apache.kafka:kafka-clients:jar:3.4.0:compile
[INFO] |  +- com.github.luben:zstd-jni:jar:1.5.2-1:runtime
[INFO] |  +- org.lz4:lz4-java:jar:1.8.0:runtime
[INFO] |  \- org.xerial.snappy:snappy-java:jar:1.1.8.4:runtime
[INFO] +- c3p0:c3p0:jar:0.9.1.2:compile
[INFO] +- org.junit.jupiter:junit-jupiter-engine:jar:5.9.3:test
[INFO] |  +- org.junit.platform:junit-platform-engine:jar:1.9.3:test
[INFO] |  |  +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  |  \- org.junit.platform:junit-platform-commons:jar:1.9.3:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-api:jar:5.9.3:test
[INFO] |  \- org.apiguardian:apiguardian-api:jar:1.1.2:test
[INFO] +- org.mockito:mockito-core:jar:4.11.0:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.12.19:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.12.19:test
[INFO] |  \- org.objenesis:objenesis:jar:3.3:test
[INFO] +- org.slf4j:slf4j-reload4j:jar:2.0.7:test
[INFO] |  \- ch.qos.reload4j:reload4j:jar:1.2.22:test
[INFO] \- org.slf4j:slf4j-api:jar:2.0.7:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.863 s
[INFO] Finished at: 2023-06-02T17:04:42+05:30
[INFO] ------------------------------------------------------------------------

```

### CODE REVIEW

File: src/test/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+                JDBCWriter jdbcWriter = new JDBCWriter(dataSource);
+
+                // Execute the method under test
+                jdbcWriter.createTable(connection, tableName, schema);
+
+                // Verify the behavior
+                verify(connection).prepareStatement("CREATE TABLE ? (?)");
+                verify(preparedStatement).execute();
+                verify(preparedStatement).close();
```
Comment (jhughes24816) [inline src/test/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriterTest.java:46]: How do these verify statements act as assertions? I may have mis understood something here

**Review by jhughes24816 — COMMENTED** [review]
Some comments but mostly looks good :)

File: src/test/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriterTest.java (hunk 1 above)
Comment (Joel-hanson) [reply src/test/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriterTest.java:46]: org.mockito.Mockito.verify Verifies certain behavior happened once.

Alias to verify(mock, times(1)) E.g:

verify(mock).someMethod("some arg");
Above is equivalent to:

verify(mock, times(1)).someMethod("some arg");

Comment (Joel-hanson) [reply src/test/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriterTest.java:46]: org.mockito.Mockito.verify Verifies certain behavior happened once.

Alias to verify(mock, times(1)) E.g:

verify(mock).someMethod("some arg");
Above is equivalent to:

verify(mock, times(1)).someMethod("some arg");

File: src/test/java/com/ibm/eventstreams/connect/jdbcsink/JDBCSinkTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        // Create a collection of SinkRecords
+        Collection<SinkRecord> records = new ArrayList<>();
+        SinkRecord record1 = new SinkRecord("topic1", 0, null, null, null, null, 0);
+        SinkRecord record2 = new SinkRecord("topic1", 1, null, null, null, null, 1);
+        records.add(record1);
+        records.add(record2);
+
+        task.database = mock(IDatabase.class);
+        Mockito.when(task.database.getWriter()).thenReturn(mock(IDatabaseWriter.class));
```
Comment (jhughes24816) [inline src/test/java/com/ibm/eventstreams/connect/jdbcsink/JDBCSinkTaskTest.java:52]: is this still supposed to be here?

**Review by jhughes24816 — COMMENTED** [review]
final comment

File: src/test/java/com/ibm/eventstreams/connect/jdbcsink/JDBCSinkTaskTest.java (hunk 2 above)
Comment (Joel-hanson) [reply src/test/java/com/ibm/eventstreams/connect/jdbcsink/JDBCSinkTaskTest.java:52]: yes, After task.start() method gets called the database gets updated with actual Databasewriter thus causing the tests to fail thus mocking the database after task.start method gets executed.

**Review by jhughes24816 — APPROVED** [review]
lgtm

**Pull request merged**